cdap-ui/server/express.js

```diff
@@ -151,6 +151,41 @@ function makeApp (authAddress, cdapConfig, uiSettings) {
 
   });
 
+  // This is used for single click deploy app from Cask Market
+  app.get('/streamApp', function(req, res) {
+    var contentLink = req.query.content;
+    var streamTo = req.query.stream;
+    var jarName = req.query.name;
+
+    var headers = {
```

Comment:
Would be more flexible if we allow UI to pass on some custom header. The only modification would be to just do `Object.query` here instead of creating a new object.